### ISSUE TITLE
restartUnits fails silently

### ISSUE THREAD

**Issue by exzombie** [Mic92/sops-nix#343]
I made a simple mistake, and specified the units without the `.service` extensions; i.e.
```
restartUnits = [ "dovecot2" ];
```
instead of
```
restartUnits = [ "dovecot2.service" ];
```
This, naturally, did not restart the service. The problem is that it did not provide any indication of the failure, either. Indicating a problem would make dealing with this kind of mistake trivial, instead of puzzling over why the passwords are not accepted. Especially because the activation script *does* say `Restarting dovecot2`, so it *really* looks like it does it, unless one goes to check the logs.

**Comment by dasJ** [comment]
It's not that easy unfortunately. We cannot really know what units exist because `systemd.packages` might pull in a couple of them without having them declaratively defined in nix

**Comment by exzombie** [comment]
Indeed, that would be difficult. However, while it would be marvelous if we could throw an error at build time, it's not really necessary to remove the biggest pain point. Which is, that the *activation script* gives no indication that the units don't exist.
